artifacts/debian/debian.go

```diff
@@ -0,0 +1,117 @@
+package debian
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/containerdisks/pkg/api"
+	"kubevirt.io/containerdisks/pkg/architecture"
+	"kubevirt.io/containerdisks/pkg/docs"
+	"kubevirt.io/containerdisks/pkg/hashsum"
+	"kubevirt.io/containerdisks/pkg/http"
+	"kubevirt.io/containerdisks/pkg/tests"
+)
+
+type debian struct {
+	Arch            string
+	Version         string
+	VersionName     string
+	getter          http.Getter
+	ExampleUserData *docs.UserData
+	envVariables    map[string]string
+}
+
+const (
+	baseURLFmt   = "https://cloud.debian.org/images/cloud/%s/latest/"
+	baseQcow2Fmt = "debian-%s-genericcloud-%s.qcow2"
+	description  = `Debian Generic Cloud images for KubeVirt.
+<br />
+<br />
+Visit [debian.org](https://cloud.debian.org/images/cloud/) to learn more about Debian project.`
+)
+
+func (d *debian) Metadata() *api.Metadata {
+	metadata := &api.Metadata{
+		Name:            "debian",
+		Version:         d.Version,
+		Description:     description,
+		ExampleUserData: *d.ExampleUserData,
+	}
+
+	if d.ExampleUserData != nil {
+		metadata.ExampleUserData = *d.ExampleUserData
+	}
+
+	return metadata
+}
+
+func (d *debian) Inspect() (*api.ArtifactDetails, error) {
+	var baseURL string
+	var qCowImageName string
+	if strings.HasPrefix(d.Version, "11") || strings.HasSuffix(d.Version, "12") {
+		baseURL = fmt.Sprintf(baseURLFmt, d.VersionName)
+		qCowImageName = fmt.Sprintf(baseQcow2Fmt, d.Version, architecture.GetImageArchitecture(d.Arch))
+	} else if strings.HasPrefix(d.Version, "13") {
+		// Debian 13 is still on "testing" phase. These builds are uploaded on daily basis to a different folder.
+		// This code should be adjusted once it reaches "stable" phase.
+		baseURL = fmt.Sprintf(baseURLFmt, d.VersionName+"/daily")
+		qCowImageName = fmt.Sprintf(baseQcow2Fmt, d.Version, architecture.GetImageArchitecture(d.Arch)+"-daily")
+	} else {
+		panic(fmt.Sprintf("can't understand provided version %s", d.Version))
```

Comment:
Please don't panic, you can return an error instead.

Reply:
Done.